### PR TITLE
navigation with filters

### PR DESCRIPTION
**Pull request by fabian202**

### CODE REVIEW

File: src/components/pages/ProgramDetailsPage/ProgramForm/index.js [hunk 1]
```diff
@@ -133,12 +135,11 @@ const ProgramForm = (props) => {
     setUpdateProgramId(program.id);
     // Fetch scholarship logs from the backend
     setLogsProgramId(program.id);
-    setCompressedParams({ pruebita: true });
+    setCompressedParams(
+      !isEmpty(queryParams) ? queryParams : { showAll: true }
```
Comment (normandanielmq) [inline src/components/pages/ProgramDetailsPage/ProgramForm/index.js:139]: Why do we need queryParams?

Comment (fabian202) [reply src/components/pages/ProgramDetailsPage/ProgramForm/index.js:139]: To filter the compressed query too, and keep the data even when the user navigate through scholarships

**Review by cesar82 — APPROVED** [review]

File: src/components/pages/ProgramDetailsPage/ProgramForm/index.js (hunk 1 above)
Comment (normandanielmq) [reply src/components/pages/ProgramDetailsPage/ProgramForm/index.js:139]: In this case, we have to use the useSearch context providers that should be saving all the search criteria

Comment (fabian202) [reply src/components/pages/ProgramDetailsPage/ProgramForm/index.js:139]: But the useSeach is not persistant, if the user refresh the page will lose the filters

Comment (normandanielmq) [reply src/components/pages/ProgramDetailsPage/ProgramForm/index.js:139]: That's correct.  However, the query could include a lot of conditions that could overflow the max length of the query string. That's why we use POST requests to the server.
It does not matter if we lose the filter after refreshing, after going back to the list filters will be cleared anyway.

**Pull request merged**